Runtime/Scripts/Export/StandardMaterialExport.cs

```diff
@@ -115,6 +123,20 @@ public override bool ConvertMaterial(UnityEngine.Material uMaterial, out Materia
                         skyboxData.skyTint = uMaterial.GetColor("_Tint");
                         skyboxData.exposure = uMaterial.GetFloat("_Exposure");
                         skyboxData.rotation = uMaterial.GetFloat("_Rotation");
+#if UNITY_EDITOR
+                        url = AssetDatabase.GetAssetPath(uMaterial.GetTexture(k_MainTex));
+                        texture = new Texture2D(2, 2);
+                        texture.LoadImage(File.ReadAllBytes(url));
+
+                        textureImporter = (TextureImporter)AssetImporter.GetAtPath(url);
+                        maxSize = textureImporter.maxTextureSize;
+                        if (texture.width > maxSize || texture.height > maxSize)
+                        {
+                            int scale = Mathf.Max(texture.width / maxSize, texture.height / maxSize);
+                            texture = ResizeTexture(texture, url, scale);
+                        }
+                        uMaterial.mainTexture = texture;
```

Comment:
Can we make this into a method so that this becomes something like
```csharp
Texture2D texture = ConvertToTexture2D(uMaterial.GetTexture(k_MainTex));
uMaterial.mainTexture = texture;
```
any relevant method name is ok but this is just an example

Reply:
I could move it all to the resize function.

Reply:
OK done